http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/sam/SamService.scala

```diff
@@ -79,6 +79,15 @@ trait SamService[F[_]] {
     ev: Ask[F, AppContext]
   ): F[String]
 
+  /**
+   * Gets a token for a user's arbitrary pet service account, using a Leonardo token.
+   * @param userEmail the user email
+   * @param ev application context
+   * @return access token for the user's arbitrary pet service account, or SamException if
+   *         the pet could not be retrieved.
+   */
+  def getArbitraryPetServiceAccountToken(userEmail: WorkbenchEmail)(implicit ev: Ask[F, AppContext]): F[String]
```

Comment:
I missed this earlier when implementing `SamService` - this is needed for Azure functionality.